lib/Widgets/IncomingTransmissionList.vala

```diff
@@ -104,6 +104,10 @@ public class Drop.Widgets.IncomingTransmissionList : Gtk.Box {
         this.show_all ();
         
         transmission_count++;
-        transmission_added (transmission);
+
+        transmission.state_changed.connect ((state) => {
+            if (state == ServerState.AWAITING_REQUEST)
+                transmission_added (transmission);
+        });
```

Comment:
It doesn't need to, signal only needs to be send in order to inform that a new one has started (e.i: Notifications). TransmissionList will still add the entries correctly.